<!-- SDU Lidar 3d demo launch file -->

<launch>
	<node pkg="sdu_lidar3d_demo" name="encoder" type="dmm_tech_abs_node" output="screen">
		<param name="device" value="/dev/ttyUSB0" />
		<param name="encoder_offset" value="4541" />
	</node>

	<node pkg="sick_lms111" name="lidar"  type="sick_lms111_node" output="screen">
		<param name="host" value="192.168.0.11" />"
		<param name="topic_id" value="/laser_lms111" />
		<param name="frame_id" value="/laser" />
	</node>

	<!-- node pkg="sdu_lidar3d_demo" name="print_pitch_roll" type="print_pitch_roll.py" output="screen">
		<param name="encoder_topic" value="/fmInformation/encoder_value" />"
	</node -->

	<node type="laser_scan_assembler" pkg="laser_assembler" name="sick_lms111_assembler">
		<remap from="scan" to="/laser_lms111"/>
		<param name="max_scans" type="int" value="400" />
		<param name="fixed_frame" type="string" value="base_link" />
	</node>

	<node type="periodic_snapshotter" pkg="laser_assembler" name="periodic_snapshots">
	</node>

	<node pkg="sdu_lidar3d_demo" name="TriangleDrawer" type="TriangleDrawer" output="screen">
		<remap from="/scan" to="/laser_lms111"/>
	</node>

	<node pkg="rviz" type="rviz" name="visualization" args="$(find sdu_lidar3d_demo)/conf/default.rviz"/>

</launch>

